cmd/rbac-manager/main.go

```diff
@@ -110,6 +112,30 @@ func createGoogleDirectory(ctx context.Context, subject string) (*directoryv1.Se
 		return nil, err
 	}
 
+	// If the found credential doesn't contain JSON, try to fallback to workload identity
+	if len(creds.JSON) == 0 {
+		// Get the email address associated with the service account. The account may be empty
+		// or the string "default" to use the instance's main account.
+		principal, err := metadata.Email("default")
+		if err != nil {
+			return nil, err
+		}
+
+		// Access to the directory API must be signed with a Subject to enable domain selection.
+		config := impersonate.CredentialsConfig{
+			TargetPrincipal: principal,
+			Scopes:          scopes,
+			Subject:         subject,
+		}
+
+		ts, err := impersonate.CredentialsTokenSource(ctx, config, option.WithCredentials(creds))
```

Comment:
Is it necessary for the account to impersonate itself, instead of using the TokenSource already returned by `FindDefaultCredentials`?

Reply:
Good question. I'm no expert in GCP auth myself and it's not very well documented or discussed anywhere, but from reading various sources, my understanding is that the federated access token obtained from the GCE
metadata server (via workload identity) is not sufficient for acting as the subject via domain-wide delegation to call Google Workspace Admin APIs.

For delegation to work, we need to sign a JWT with the the "sub" claim set to subject - this happens implicitly through impersonation.

You can kind of this behaviour in `userTokenSource`: https://github.com/googleapis/google-api-go-client/blob/af0a938ba1e01e7c2ce7f96fef1e62478f5da784/impersonate/user.go#L96-L103

Whereas you don't see this in `computeSource` obtained by WID: https://cs.opensource.google/go/x/oauth2/+/refs/tags/v0.24.0:google/google.go;l=270

Reply:
Also, the changes in this PR were largely inspired / adapted from https://github.com/dexidp/dex/pull/2989

Reply:
Some useful reading:
- https://blog.salrashid.dev/articles/2021/impersonation_and_domain_delegation
- https://cloud.google.com/iam/docs/workload-identity-federation#access_management

Reply:
That said, looks like there have been some new developments since I first opened this PR.. 👀 

It looks like it's possible to assign Workspace Admin `Groups Reader` role to service accounts:
- https://support.google.com/a/answer/9807615?sjid=13011041076049488223-AP#:~:text=Assign%20a%20role%20to%20a%20service%20account
- https://cloud.google.com/identity/docs/how-to/setup#auth-no-dwd

Which might mean that there's no need for impersonation or domain-wide delegation anymore, which would greatly simplify the auth here (just `google.FindDefaultCredentials` might be all that's needed) 🤔

But I think that warrants a separate PR as it's a different feature altogether..